lib/ExternalModuleFactoryPlugin.js

```diff
@@ -27,10 +27,23 @@ class ExternalModuleFactoryPlugin {
 					}
 					if (value === false) return factory(data, callback);
 					if (value === true) value = dependency.request;
-					if (type === undefined && /^[a-z0-9]+ /.test(value)) {
-						const idx = value.indexOf(" ");
-						type = value.substr(0, idx);
-						value = value.substr(idx + 1);
+					if (type === undefined) {
+						if (typeof value === "string" && /^[a-z0-9]+ /.test(value)) {
+							const idx = value.indexOf(" ");
+							type = value.substr(0, idx);
+							value = value.substr(idx + 1);
+						} else if (typeof value === "object" && !Array.isArray(value)) {
+							const keys = Object.keys(value);
```

Comment:
This case is weird. There is no special handling for single key objects in the ExternalModule.

Instead check in ExternalModule, if the accessed key is not undefined:

https://github.com/webpack/webpack/blob/df7da2cdd58e9534ae5e73a25e9c064b02c4d625/lib/ExternalModule.js#L116-L119

``` js
if(!request) {
  throw new Error(...);
}
```